csv/result_test.go

```diff
@@ -467,6 +468,16 @@ func TestResultDecoder(t *testing.T) {
 				}},
 			},
 		},
+		{
+			name:          "simple error message",
+			encoderConfig: csv.DefaultEncoderConfig(),
+			encoded: toCRLF(`error,reference
```

Comment:
To decode it is required that the csv be annotated. Can you make these tests use the `annotated` versions of csv encoding. Then it will be clear that `error` is not an annotation but rather a column header.

Reply:
I've updated things so the error is now encoded with annotations and I decode that now instead.

It seems that the decoder has a `NoHeader` attribute that changes how things work, but it seems intended to mean no header column rather than no annotations so it doesn't appear there's a way to detect that there were no annotations.